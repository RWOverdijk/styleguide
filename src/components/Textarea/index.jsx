import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import checkIcon from '../../assets/icons/fa/check.svg';
import closeIcon from '../../assets/icons/fa/close.svg';
import Icon from '../Icon';
import './style.scss';

const Textarea = ({ meta, input, className, children, ...props }) => {
  const { error, warning, touched } = meta;
  const classNames = classnames(
    'Van-TextInput',
    className,
    { 'Van-TextInput--error': touched && error },
    { 'Van-TextInput--warning': touched && warning },
    { 'Van-TextInput--success': touched && !error && !warning }
  );

  return (
    <div className={classNames}>
      <textarea className="Van-TextInput-input" {...props} {...input}>
        {children}
      </textarea>

      {touched && error && <Icon source={closeIcon} className="Van-TextInput-icon" />}

      {touched && !error && !warning && <Icon source={checkIcon} className="Van-TextInput-icon" />}

      {touched &&
        typeof error === 'string' && <div className="Van-TextInput-error-message">{error}</div>}
    </div>
  );
};

Textarea.propTypes = {
  meta: PropTypes.object,
  input: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.string
};

Textarea.defaultProps = {
  meta: {},
  input: {},
  className: '',
  children: ''
};

export default Textarea;
