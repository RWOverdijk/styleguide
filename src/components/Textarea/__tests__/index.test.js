import React from 'react';
import renderer from 'react-test-renderer';
import Textarea from '../';

describe('<Textarea/>', () => {
  it("renders error input", () => {
    const tree = renderer.create(
      <Textarea meta={{ touched: true, error: true }}>
        hello
      </Textarea>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders error input with error message", () => {
    const tree = renderer.create(
      <Textarea meta={{ touched: true, error: 'My error message' }}>
        hello
      </Textarea>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders success input", () => {
    const tree = renderer.create(
      <Textarea meta={{ touched: true }}>
        hello
      </Textarea>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Textarea className="my-class"></Textarea>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Textarea data-id="0"></Textarea>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
