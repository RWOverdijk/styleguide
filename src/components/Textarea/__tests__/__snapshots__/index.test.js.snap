// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea/> renders error input 1`] = `
<div
  className="Van-TextInput Van-TextInput--error"
>
  <textarea
    className="Van-TextInput-input"
  >
    hello
  </textarea>
</div>
`;

exports[`<Textarea/> renders error input with error message 1`] = `
<div
  className="Van-TextInput Van-TextInput--error"
>
  <textarea
    className="Van-TextInput-input"
  >
    hello
  </textarea>
  <div
    className="Van-TextInput-error-message"
  >
    My error message
  </div>
</div>
`;

exports[`<Textarea/> renders success input 1`] = `
<div
  className="Van-TextInput Van-TextInput--success"
>
  <textarea
    className="Van-TextInput-input"
  >
    hello
  </textarea>
</div>
`;

exports[`<Textarea/> renders with specific attributes 1`] = `
<div
  className="Van-TextInput"
>
  <textarea
    className="Van-TextInput-input"
    data-id="0"
  >
    
  </textarea>
</div>
`;

exports[`<Textarea/> renders with specific className 1`] = `
<div
  className="Van-TextInput my-class"
>
  <textarea
    className="Van-TextInput-input"
  >
    
  </textarea>
</div>
`;
